pandas/tests/groupby/test_categorical.py

```diff
@@ -1193,3 +1193,14 @@ def test_groupby_categorical_axis_1(code):
     result = df.groupby(cat, axis=1).mean()
     expected = df.T.groupby(cat, axis=0).mean().T
     assert_frame_equal(result, expected)
+
+
+def test_groupby_cat_preserves_group_name():
+    # GH 28787
+    df = DataFrame({"a": [1, 2, 3]})
+    expected = df.groupby("a").grouper.levels[0].name
+
+    df["a"] = df["a"].astype("category")
+    result = df.groupby("a").grouper.levels[0].name
+
+    assert result == expected
```

Comment:
You should also test that the result is actually ``"a"``, in case there a bug that also affects the expected groupby.

Reply:
Done